rules/java-net/src/main/resources/otagent/java-net.btm

```diff
@@ -0,0 +1,110 @@
+
+RULE java-net: Java HttpUrlConnection connect entry
+CLASS ^java.net.HttpURLConnection
+METHOD connect()
+HELPER io.opentracing.contrib.agent.OpenTracingHelper
+AT ENTRY
+IF !$0.connected && retrieveSpan($0) == null && !ignore($0)
+DO
+  associateSpan($0, getTracer().buildSpan($0.getRequestMethod())
```

Comment:
I wonder if there's a way to provide some mechanism for people to control span names. Some (possibly dumb ideas):
  * use some fixed http header like `endpoint`
  * use some annotations in the code

the best solution is to recommend people to use RPC frameworks for communications rather than low-level networking libs.